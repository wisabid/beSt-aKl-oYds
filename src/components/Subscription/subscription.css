.bao-subscription {
    color : slateblue;
    font-weight: bold;
    font-style: italic;
    cursor: pointer;
}
.e_subsbtn
{
    padding: 5px;
    font-style: italic;
    width: 100px;
}
.m_subsbtn
{
    padding: 3px;
    font-size: 1.7vmin;
    font-style: italic;
    width: 80px;
}
.o_subsbtn
{
    padding: 3px;
    font-size: 1.5vmin;
    font-style: italic;
    width: 70px;
}