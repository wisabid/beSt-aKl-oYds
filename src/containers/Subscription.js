import { connect } from 'react-redux';
import SubsComp from '../components/Subscription';
import {subscribe_bao, unsubscribe_bao} from '../store/actions/'

const mapStateToProps = (state) => {
    return {        
        subscriptions : state.rL.subscriptions
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        subscribe_event : (payload) => dispatch(subscribe_bao(payload)),
        unsubscribe_event : (payload) => dispatch(unsubscribe_bao(payload)),
    }
}

const Subscription = connect(mapStateToProps, mapDispatchToProps)(SubsComp);

export default Subscription;