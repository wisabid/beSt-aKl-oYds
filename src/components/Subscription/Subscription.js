import React from 'react';
import './subscription.css'

const Subscription = ({subscriptions, uid, subscribe_event, unsubscribe_event}) => {
    debugger;
    return (
        <div className="bao-subscription">
        {(subscriptions.indexOf(uid) === -1)
            ?<button className={`${uid.substr(0,1)}_subsbtn`} onClick={() => subscribe_event(uid)}>Subscribe</button>
            :<button className={`${uid.substr(0,1)}_subsbtn`} onClick={() => unsubscribe_event(uid)} style={{background:"Orange"}}>Unsubscribe</button>   
        }
        </div>                         
    )
}

export default Subscription;