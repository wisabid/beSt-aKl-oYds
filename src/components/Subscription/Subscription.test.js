import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Subscription from './Subscription';

jest.mock('../../context/UserContext');

describe('<Subscription />', () => {
  const SubscriptionComp = shallow(<Subscription subscriptions={[]} uid="1234" />)
  it('<Subscription /> renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Subscription subscriptions={[]} uid="1234"/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('<Subscription /> component matches the snapshot', () => {
    expect(SubscriptionComp).toMatchSnapshot();
  })

  

});
