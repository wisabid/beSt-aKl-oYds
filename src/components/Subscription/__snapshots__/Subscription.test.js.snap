// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subscription /> <Subscription /> component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Subscription
    subscriptions={Array []}
    uid="1234"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="1_subsbtn"
        onClick={[Function]}
        title="Subscribe to 1234"
      >
        Subscribe
      </button>,
      "className": "bao-subscription",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Subscribe",
        "className": "1_subsbtn",
        "onClick": [Function],
        "title": "Subscribe to 1234",
      },
      "ref": null,
      "rendered": "Subscribe",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="1_subsbtn"
          onClick={[Function]}
          title="Subscribe to 1234"
        >
          Subscribe
        </button>,
        "className": "bao-subscription",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Subscribe",
          "className": "1_subsbtn",
          "onClick": [Function],
          "title": "Subscribe to 1234",
        },
        "ref": null,
        "rendered": "Subscribe",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
